pkg/plugin/volume.go

```diff
@@ -85,3 +115,48 @@ func buildVolumeMount(bucket string, mountPath string) *corev1.VolumeMount {
 func hostPathTypePtr(v corev1.HostPathType) *corev1.HostPathType {
 	return &v
 }
+
+// ensurePVC creates a PVC based on the config present in the backupstoragelocation CRD
+func ensurePVC(config map[string]string, log *logrus.Entry) error {
+	namespace := os.Getenv("VELERO_NAMESPACE")
+
+	clientset, err := k8sutil.GetClientset()
+	if err != nil {
+		return errors.Wrap(err, "failed to get clientset")
+	}
+
+	pvcObj, err := clientset.CoreV1().PersistentVolumeClaims(namespace).Get(context.TODO(), config["bucket"], metav1.GetOptions{})
+	if err != nil && !kuberneteserrors.IsNotFound(err) {
+		return errors.Wrap(err, "failed to get velero pvc")
+	}
+	if err == nil {
+		log.Infof("pvc already exists: %s", pvcObj.Name)
+		return nil
+	}
+
+	persistentVolumeClaim := &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: config["bucket"],
+			Labels: map[string]string{
+				VolumeProviderKey: VolumeProviderLabel,
+			},
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			AccessModes: []corev1.PersistentVolumeAccessMode{
+				corev1.ReadWriteMany,
+			},
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceName(corev1.ResourceStorage): resource.MustParse(config["storageSize"]),
+				},
+			},
+		},
```

Comment:
I think I just realized the reason this worked on the longhorn cluster with the wrong `storageClassName` is because there is nowhere we use it. We'll need it to make sure we use the RETAIN storage class.. We should also support the default case.

Reply:
@DanStough you may already know this. Just writing this for my own memory.  Use PV's Policy to Retain without touching StorageClass and PVC. By tweaking one of the Provisioners' StorageClass, say longhorn, we need to tweak the StorageClass for every PVC Provisioner say Rook.

Reply:
I have also verified if the backup works fine if the default StorageClass Policy is set to Retain.